client/state/slices/userSlice.ts

```diff
@@ -0,0 +1,31 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+import { RootState } from "../store";
+import { CurrentUser } from "@/types/userTypes";
+
+const initialState: CurrentUser = {};
+
+const userSlice = createSlice({
+  name: "user",
+  initialState,
+  reducers: {
+    setUser: (_, action: PayloadAction<CurrentUser>) => ({
+      ...action.payload,
+    }),
+    setUsername: (state, action: PayloadAction<string>) => {
+      state.username = action.payload;
+    },
+    clearUser: () => initialState,
+  },
+});
+
+export const selectCurrentUserId = (state: RootState) => state.user.userId;
+
+export const selectIsUserSignedIn = (state: RootState): boolean =>
+  typeof state.user.userId === "string" && state.user.userId.length > 0;
```

Comment:
Do we need this `typeof` check? Also, if we're checking for whether the string is empty or not I think something like this also works but is simpler

```suggestion
  !!state.user.userId
```

Reply:
Good catch, I just had that typeof check because it shouldn't be undefined, but I think that is covered with the suggested change as a falsy value